src/pkg/backup/details/chats.go

```diff
@@ -0,0 +1,120 @@
+package details
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/alcionai/clues"
+
+	"github.com/alcionai/corso/src/pkg/dttm"
+	"github.com/alcionai/corso/src/pkg/path"
+)
+
+// NewChatsLocationIDer builds a LocationIDer for the chats.
+func NewChatsLocationIDer(
+	category path.CategoryType,
+	escapedFolders ...string,
+) (uniqueLoc, error) {
+	if err := path.ValidateServiceAndCategory(path.TeamsChatsService, category); err != nil {
+		return uniqueLoc{}, clues.Wrap(err, "making chats LocationIDer")
+	}
+
+	pb := path.Builder{}.Append(category.String()).Append(escapedFolders...)
+
+	return uniqueLoc{
+		pb:          pb,
+		prefixElems: 1,
+	}, nil
+}
+
+// TeamsChatsInfo describes a chat within teams chats.
+type TeamsChatsInfo struct {
+	ItemType   ItemType  `json:"itemType,omitempty"`
+	Modified   time.Time `json:"modified,omitempty"`
+	ParentPath string    `json:"parentPath,omitempty"`
+
+	Chat ChatInfo `json:"chat,omitempty"`
+}
+
+type ChatInfo struct {
+	CreatedAt          time.Time `json:"createdAt,omitempty"`
+	HasExternalMembers bool      `json:"hasExternalMemebers,omitempty"`
+	LastMessageAt      time.Time `json:"lastMessageAt,omitempty"`
+	LastMessagePreview string    `json:"preview,omitempty"`
+	Members            []string  `json:"members,omitempty"`
+	MessageCount       int       `json:"size,omitempty"`
+	Name               string    `json:"name,omitempty"`
+}
+
+// Headers returns the human-readable names of properties in a ChatsInfo
+// for printing out to a terminal in a columnar display.
+func (i TeamsChatsInfo) Headers() []string {
+	switch i.ItemType {
+	case TeamsChat:
+		return []string{"Name", "Last message", "Last message at", "Message count", "Created", "Members"}
+	}
+
+	return []string{}
+}
+
+// Values returns the values matching the Headers list for printing
+// out to a terminal in a columnar display.
+func (i TeamsChatsInfo) Values() []string {
+	switch i.ItemType {
+	case TeamsChat:
+		members := ""
+		icmLen := len(i.Chat.Members)
+
+		if icmLen > 0 {
+			members = i.Chat.Members[0]
+		}
+
+		if icmLen > 1 {
+			members = fmt.Sprintf("%s, and %d more", members, icmLen-1)
+		}
+
+		return []string{
+			i.Chat.Name,
+			i.Chat.LastMessagePreview,
+			dttm.FormatToTabularDisplay(i.Chat.LastMessageAt),
+			strconv.Itoa(i.Chat.MessageCount),
```

Comment:
Curious, what's the value add for displaying message count?

Reply:
Users don't have much data to work with when reviewing details.  The chat topic (ie, it's name) is often unpopulated.  Multiple chats can contain the same people.  Date times only go so far.  Personally, I've found that looking for "that chat where I had a lo of messages" is one of the more effective associations.

That's probably a more personal thing, because they're my chats personally.  But I figure it can't hurt either way.

Reply:
I see. Yeah, that's good reasoning. Thanks!